engine/src/test/java/org/camunda/bpm/engine/test/api/history/removaltime/RemovalTimeStrategyEndTest.java

```diff
@@ -349,6 +353,92 @@ public void shouldResolveHistoricTaskInstance() {
     assertThat(historicTaskInstance.getRemovalTime(), is(removalTime));
   }
 
+  @Test
+  public void shouldWriteHistoryAndResolveHistoricTaskAuthorizationInDifferentTransactions() {
+    // given
+    processEngineConfiguration.setEnableHistoricInstancePermissions(true);
+
+    testRule.deploy(CALLING_PROCESS);
+
+    testRule.deploy(CALLED_PROCESS);
+
+    ClockUtil.setCurrentTime(START_DATE);
+
+    runtimeService.startProcessInstanceByKey(CALLING_PROCESS_KEY);
+
+    String taskId = taskService.createTaskQuery().singleResult().getId();
+
+    enabledAuth();
+    taskService.setAssignee(taskId, "myUserId");
+    disableAuth();
+
+    Authorization authorization = authorizationService.createAuthorizationQuery()
+        .resourceType(Resources.HISTORIC_TASK)
+        .singleResult();
+
+    // assume
+    assertThat(authorization.getRemovalTime(), nullValue());
+
+    ClockUtil.setCurrentTime(END_DATE);
+
+    // when
+    taskService.complete(taskId);
+
+    authorization = authorizationService.createAuthorizationQuery()
```

Comment:
Nitpicking: The `//then` part of the test begins here. For me, the most important aspect of the `given-when-then` structure is that I can quickly understand which functionality is tested, so the `when` part should contain only the statements that trigger the code under test.

Reply:
My understanding is the following:
* `when` refers to the next line of code
* `then` refers to the assertions
* code lines between `when+ 1` and `then` retrieve data in preparation of the assertions

Reply:
Okay, I see the role of `then` differently. It should include the code that fetches the data. I can already find assertions easily by looking for `assert*` lines. Maybe we can discuss this in the next team meeting and clarify our testing best practice. I don't mind if we merge this PR in one way or the other.

Reply:
👍